packages/client/src/userContext.js

```diff
@@ -0,0 +1,70 @@
+import React, {
+  createContext,
+  useContext,
+  useState,
+  useMemo,
+  useEffect,
+  useCallback,
+} from 'react';
+import PropTypes from 'prop-types';
+import { useAuthState } from 'react-firebase-hooks/auth';
+import {
+  auth,
+  db,
+  logInWithEmailAndPassword,
+  signInWithGoogle,
+  registerWithEmailAndPassword,
+  logout,
+} from './firebase';
+import { query, collection, getDocs, where } from 'firebase/firestore';
+
+export const UserContext = createContext();
```

Comment:
Nitpick: Not sure you really want to export this, since the gain of custom hook and custom provider is to not expose this detail.